assets/src/editor/blocks/share-article/index.js

```diff
@@ -37,36 +42,122 @@ export const settings = {
 			type: 'boolean',
 			default: true,
 		},
+		enableLinkedIn: {
+			type: 'boolean',
+			default: true,
+		},
+		enableEmail: {
+			type: 'boolean',
+			default: true,
+		},
+		enableCopyLink: {
+			type: 'boolean',
+			default: true,
+		},
 	},
 
 	example: {},
 
 	/**
-	 * Edit component used to manage featured image and page intro.
+	 * Edit component for managing social share settings.
 	 */
 	edit: function ShareArticleBlock( { attributes, setAttributes } ) {
 		const {
 			enableTwitter,
 			enableFacebook,
+			enableLinkedIn,
+			enableEmail,
+			enableCopyLink,
 		} = attributes;
 
 		const blockProps = useBlockProps( {
 			className: 'share-article',
 		} );
 
 		return (
-			<div { ...blockProps } >
-				{ ( ! enableTwitter && ! enableFacebook ) && (
+			<div { ...blockProps }>
+				{ ( ! enableTwitter && ! enableFacebook && ! enableLinkedIn && ! enableEmail && ! enableCopyLink ) && (
 					<small>{ __( '(No social share will be shown)', 'shiro-admin' ) }</small>
 				) }
-				<div className="share-article">
-					{ enableTwitter && ( <div className="share-article__link">
-						<TwitterIcon />
-					</div> ) }
-
-					{ enableFacebook && ( <div className="share-article__link">
-						<FacebookIcon />
-					</div> ) }
+					<div className="share-button-container share-article">
```

Comment:
Nitpick, lost a level of indentation on the wrapper `div`.
```suggestion
				<div className="share-button-container share-article">
```